validator/src/storage/mod.rs

```diff
@@ -33,3 +30,31 @@ pub trait BlockStore {
 
     fn iter(&self) -> Box<Iterator<Item = Block>>;
 }
+
+#[derive(Debug, PartialEq)]
+pub enum BlockManagerError {
+    MissingPredecessor(String),
+    MissingInput,
+}
+
+pub trait BlockManager {
```

Comment:
Are there going to be multiple implementations of the BlockManager?

Reply:
I was thinking that the unsync and the sync versions of the block manager would both implement the trait. I agree that it is probably best to not have it be a trait.